Mathlib/Analysis/Meromorphic/NormalFormAt.lean

```diff
@@ -135,6 +124,151 @@ theorem AnalyticAt.meromorphicNFAt (hf : AnalyticAt 𝕜 f x) :
     MeromorphicNFAt f x := by
   simp [meromorphicNFAt_iff_analyticAt_or, hf]
 
+/-- Meromorphic functions have normal form outside of a discrete subset in the domain of
+meromorphicity. -/
+theorem MeromorphicOn.meromorphicNFAt_codiscreteWithin [CompleteSpace E] {U : Set 𝕜}
+    (hf : MeromorphicOn f U) :
+    { x | MeromorphicNFAt f x } ∈ Filter.codiscreteWithin U := by
+  filter_upwards [hf.analyticAt_codiscreteWithin] with _ ha
+  exact ha.meromorphicNFAt
+
+/- Private helper lemma. -/
```

Comment:
Note that mathlib requires that all *definitions* come with a docstring. Theorems/lemmas do not have to be documented, but it is fine to do so if the docstring helps understanding the statement (or provides a reference, etc...).
```suggestion
```